js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+            maxRequestedMilliSeconds = maxAllowedDiapason * 1000;
+        } else {
+            maxRequestedMilliSeconds = Math.min (this.sum (limit, 1) * duration, maxAllowedDiapason) * 1000;
+        }
+        let response = null;
+        if (exchangeType === 'spot') {
+            request['interval'] = this.timeframes[timeframe];
+            const now = this.milliseconds ();
+            let startTime = since;
+            let endTime = now + 1000; // At first, Let's add one second
+            if (startTime === undefined) {
+                startTime = endTime - maxRequestedMilliSeconds;
+            } else {
+                endTime = this.sum (startTime, maxRequestedMilliSeconds);
+            }
+            startTime = startTime / 1000;
+            endTime = endTime / 1000;
+            request['start'] = Math.ceil (startTime);
+            request['end'] = Math.ceil (endTime);
+            response = await this.spotPublicGetKline (this.extend (request, params));
+            // {
+            //     success: true,
+            //     message: "",
+            //     result: {
+            //         market: "BTC_USDT",
+            //         start: 1640000000,
+            //         end: 1640001234,
+            //         interval: 60,
+            //         kline: [
+            //             {
+            //                 time: 1640000000,
+            //                 open: "46979.18",
+            //                 close: "46893.88",
+            //                 highest: "46979.18",
+            //                 lowest: "46893.88",
+            //                 volume: "2.35947525",
+            //                 amount: "110745.7371631963",
+            //                 market: "BTC_USDT"
+            //             },
+            //             {
+            //                 time: 1640000060,
+            //                 ....
+        } else if (exchangeType === 'futures') {
+            response = [];
+        }
+        if (exchangeType === 'spot') {
+            const result = this.safeValue (response, 'result', {});
+            const data = this.safeValue (result, 'kline', []);
+            return this.parseOHLCVs (data, market, timeframe, since, limit);
+        }
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        // the ordering in spot candles is OCHLV
+        //
+        //     {
+        //         time: '1640051700',
+        //         open: '46954.3',
+        //         close: '46932.07',
+        //         highest: '46960.75',
+        //         lowest: '46932.07',
+        //         volume: '1.6985718',
+        //         amount: '79753.7310157163',
+        //         market: 'BTC_USDT'
+        //     }
+        //
+        // the ordering in swap / contract candles is OHLCV
+        //
+        if (market === undefined) {
+            const symbol = this.safeValue (ohlcv, 'market');
+            market = this.market (symbol);
```

Comment:
Let's rewrite the two lines above in the following manner for less confusion:

```JavaScript
const marketId = this.safeString (ohlcv, 'market');
market = this.market (marketId, market);
```

Reply:
yeah, much better.

Reply:
Sorry, that's wrong. Please do this:
```JavaScript
const marketId = this.safeString (ohlcv, 'market');
market = this.safeMarket (marketId, market);
```

Reply:
If they have a distinct symbol delimiter, we can add that as the third argument to `safeMarket`.